sdk/storage/storage-blob/src/Lease.ts

```diff
@@ -0,0 +1,173 @@
+import * as Models from "../src/generated/lib/models";
+import { Aborter } from "./Aborter";
+import { ContainerClient } from "./ContainerClient";
+import { Blob, Container } from "./generated/lib/operations";
+import { StorageClientContext } from "./generated/lib/storageClient";
+import { BlobClient } from "./internal";
+
+/**
+ * Configures lease operations.
+ *
+ * @export
+ * @interface LeaseOperationOptions
+ */
+export interface LeaseOperationOptions {
+  /**
+   * Aborter instance to cancel request. It can be created with Aborter.none
+   * or Aborter.timeout(). Go to documents of {@link Aborter} for more examples
+   * about request cancellation.
+   *
+   * @type {Aborter}
+   * @memberof LeaseOperationOptions
+   */
+  abortSignal?: Aborter;
+  /**
+   * Conditions to meet when changing the lease.
+   *
+   * @type {Models.ModifiedAccessConditions}
+   * @memberof LeaseOperationOptions
+   */
+  modifiedAccessConditions?: Models.ModifiedAccessConditions;
+}
+
+/**
+ * A client that manages leases for a ContainerClient or a BlobClient.
+ *
+ * @export
+ * @class LeaseClient
+ */
+export class LeaseClient {
+
+  public get leaseId() {
+    return this._leaseId;
+  }
+
+  /**
+   * To end the lease but ensure that another client cannot acquire a new lease
+   * until the current lease period has expired.
+   * @see https://docs.microsoft.com/en-us/rest/api/storageservices/lease-container
+   * and
+   * @see https://docs.microsoft.com/en-us/rest/api/storageservices/lease-blob
+   *
+   * @static
+   * @param {(ContainerClient | BlobClient)} client
+   * @param {number} breakPeriod Break period
+   * @param {LeaseOperationOptions} [options={}] Optional options to configure lease management operations.
+   * @memberof LeaseClient
+   */
+  public static async breakAll(
+    client: ContainerClient | BlobClient,
+    breakPeriod: number,
+    options: LeaseOperationOptions = {}) {
+    const aborter = options.abortSignal || Aborter.none;
+    const operationOptions = {
+      abortSignal: aborter,
+      breakPeriod,
+      modifiedAccessConditions: options.modifiedAccessConditions
+    };
+    const clientContext = new StorageClientContext(client.url, client.pipeline.toServiceClientOptions());
+    if (client instanceof ContainerClient) {
+      const container = new Container(clientContext);
+      await container.breakLease(operationOptions);
+    } else {
+      const blob = new Blob(clientContext);
+      await blob.breakLease(operationOptions);
+    }
+  }
+
+  /**
+   * Creates an instance of LeaseClient.
+   * @param {string} _leaseId Initial proposed lease id.
+   * @param {(Container | Blob)} containerOrBlob The object that is used to make the lease operation requests.
```

Comment:
Is this better described as the "scope" of the lease or some such (i.e. WHAT you are leasing)?

Reply:
This parameter is the object from the protocol layer which is used to make the service call.  Yeah I struggled to find a better description.

Reply:
I changed the signature to take in resource type and storageclientcontext instead. Seems better.